pipeline_steps/common.groovy

```diff
@@ -1969,3 +1974,126 @@ void updateStringParam(String name, String value, String description){
     ParametersAction newAction = existingAction.createUpdated(updatedParam)
     currentBuild.rawBuild.addOrReplaceAction(newAction)
 }
+
+/**
+* Get the predicted duration for the current build in milliseconds.
+* Prediction is based on up to 6 recent builds.
+* See: https://javadoc.jenkins-ci.org/hudson/model/Job.html#getEstimatedDurationCandidates--
+* NonCPS because build is not serialisable
+*/
+@NonCPS
+Duration getPredictedDuration(){
```

Comment:
I like, it's clean.  But would be more 'functional' or re-usable if we passed in the parent job as an input parameter. Meh, either way works.  This is clean enough.  Just a thought. Or perhaps renaming to getCurrentBuildPredictedDuration() to properly describe what it does, but that's too long of a method name.  Keep it as-is - just thinking out loud.